File: chap3/Initiales.java
// Ces programmes sont sous licence CeCILL-B V1.
class Initiales {

  static public void main (String [] Args) {
    String nom;
    int i;
    nom = Isn.readString();
    System.out.print(Isn.stringNth(nom,0));
    for (i = 0; i <= Isn.stringLength(nom) - 2; i = i + 1) {
      if (Isn.stringEqual(" ", Isn.stringNth(nom, i))) {
        System.out.print(Isn.stringNth(nom, i + 1));}}
    System.out.println();}}

File: chap3/Initiales.ml
(* Ces programmes sont sous licence CeCILL-B V1. *)
let _ =
  let nom = input_line stdin in
  print_char nom.[0];
  for i = 1 to String.length nom - 2 do
    if nom.[i] = ' ' then
      print_char nom.[i + 1]
  done;
  print_newline ()
